src/Evolution/Systems/GeneralizedHarmonic/BoundaryConditions/BjorhusImpl.cpp

```diff
@@ -0,0 +1,215 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Evolution/Systems/GeneralizedHarmonic/BoundaryConditions/BjorhusImpl.hpp"
+
+#include <algorithm>
+
+#include "Utilities/GenerateInstantiations.hpp"
+
+namespace GeneralizedHarmonic::BoundaryConditions::Bjorhus {
+
+template <size_t VolumeDim, typename DataType>
+void add_constraint_preserving_terms_to_dt_v_minus(
+    const gsl::not_null<tnsr::aa<DataType, VolumeDim, Frame::Inertial>*>
+        bc_dt_v_minus,
+    const Scalar<DataType>& gamma2,
+    const tnsr::I<DataType, VolumeDim, Frame::Inertial>& inertial_coords,
+    const tnsr::a<DataType, VolumeDim, Frame::Inertial>& incoming_null_one_form,
+    const tnsr::a<DataType, VolumeDim, Frame::Inertial>& outgoing_null_one_form,
+    const tnsr::A<DataType, VolumeDim, Frame::Inertial>& incoming_null_vector,
+    const tnsr::A<DataType, VolumeDim, Frame::Inertial>& outgoing_null_vector,
+    const tnsr::aa<DataType, VolumeDim, Frame::Inertial>& projection_ab,
+    const tnsr::Ab<DataType, VolumeDim, Frame::Inertial>& projection_Ab,
+    const tnsr::AA<DataType, VolumeDim, Frame::Inertial>& projection_AB,
+    const tnsr::aa<DataType, VolumeDim, Frame::Inertial>&
+        char_projected_rhs_dt_v_psi,
+    const tnsr::aa<DataType, VolumeDim, Frame::Inertial>&
+        char_projected_rhs_dt_v_minus,
+    const tnsr::a<DataType, VolumeDim, Frame::Inertial>&
+        constraint_char_zero_plus,
+    const tnsr::a<DataType, VolumeDim, Frame::Inertial>&
+        constraint_char_zero_minus,
+    const std::array<DataType, 4>& char_speeds) noexcept {
+  if (UNLIKELY(get_size(get<0, 0>(*bc_dt_v_minus)) != get_size(get(gamma2)))) {
+    *bc_dt_v_minus =
```

Comment:
The value being set here doesn't matter since the following code resets it to `-1 x` the current time-derivative and then adds the full RHS, leaving behind only the corrections to `dt<VMinus>` as intended.

Regarding the size: if a pointer to a default constructed object is passed, this function will allocate memory of correct size and fill it in correctly. I think that is more reasonable behavior than an assertion. Any reason why you think assertion would still be better (now that value overwriting is not an issue)?

Reply:
We don't want to destructive resize here as the pointer passed to this impl function would typically be pointing into a TempBuffer, and not to an owning tensor, and so cannot be readily destructive_resized (unless I misunderstand how that works). This is why its probably better to reallocate a new memory, similar to [here](https://github.com/sxs-collaboration/spectre/pull/2998#discussion_r633537848).

Reply:
Ah, but if this points to a non-owning structure, we also can't assign it a different size with the `make_with_values` as you're doing here. If it's non-owning it'll generate an error either way, so my opinion would be that you should probably either assert the size is right or use `destructive_resize`.

Reply:
I agree, and if this is overwriting things to `-1x` then the function name is a bit misleading

Reply:
Oops yes, changed to using `destructive_resize`. Also renamed this function to `constraint_preserving_bjorhus_corrections_dt_v_minus` to conform to the convention of similar functions' names for `v_psi` and `v_zero`.